src/sentry/static/sentry/app/components/group/suggestedOwners/suggestedOwners.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+
+import {assignToUser, assignToActor} from 'app/actionCreators/group';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+import Access from 'app/components/acl/access';
+import {Organization, Group, Event, Actor, Commit, Project, User} from 'app/types';
+import {Client} from 'app/api';
+
+import {findMatchedRules, Rules} from './findMatchedRules';
+import {SuggestedAssignees} from './suggestedAssignees';
+import {OwnershipRules} from './ownershipRules';
+
+type OwnerList = React.ComponentProps<typeof SuggestedAssignees>['owners'];
+
+type Committer = {
+  author: User;
+  commits: Array<Commit>;
+};
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  project: Project;
+  group: Group;
+  event: Event;
+};
+
+type State = {
+  rules: Rules;
+  owners: Array<Actor>;
+  committers: Array<Committer>;
+};
+
+class SuggestedOwners extends React.Component<Props, State> {
+  state: State = {
+    rules: null,
+    owners: [],
+    committers: [],
+  };
+
+  componentDidMount() {
+    this.fetchData(this.props.event);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.event && prevProps.event) {
+      if (this.props.event.id !== prevProps.event.id) {
+        //two events, with different IDs
+        this.fetchData(this.props.event);
+      }
+      return;
+    }
+
+    if (this.props.event) {
+      //going from having no event to having an event
+      this.fetchData(this.props.event);
+    }
+  }
+
+  async fetchData(event: Event) {
+    if (!event) {
+      return;
+    }
+
+    // No committers if you don't have any releases
+    if (!!this.props.group.firstRelease) {
+      this.fetchCommitters(event.id);
+    }
+
+    this.fetchOwners(event.id);
+  }
+
+  fetchCommitters = async (eventId: Event['id']) => {
+    const {api, project, organization} = this.props;
+    // TODO: move this into a store since `EventCause` makes this exact request as well
+    await api
+      .requestPromise(
+        `/projects/${organization.slug}/${project.slug}/events/${eventId}/committers/`
+      )
+      .then(data => {
+        this.setState({
+          committers: data.committers,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          committers: [],
+        });
+      });
```

Comment:
Let's not mix await with then/catch

Reply:
thank you for the tip Billy... Code is updated 😉